packages/map-template/src/components/Search/Categories/Categories.jsx

```diff
@@ -141,29 +153,61 @@ function Categories({ onSetSize, getFilteredLocations, searchFieldRef, isOpen })
         }
     }, [activeCategory, category, isBottomSheetLoaded]);
 
-    /**
-     * Collects all child keys from all categories.
-     * Filters out categories from the initial view that are listed as a child to another category.
+    /*
+     * Filters and sets the categories to display based on whether top-level or sub-categories should be shown.
+     * If `topLevelCategory` is true, it excludes categories that appear as children.
+     * Otherwise, it includes only categories that are children.
      */
     useEffect(() => {
         const childKeys = categories.flatMap(([, category]) => category.childKeys || []);
-        const topLevelCategories = categories.filter(([key]) => !childKeys.includes(key));
-        setCategoriesWithoutChildKeys(topLevelCategories);
-    }, [categories]);
+        const categoriesToDisplay = topLevelCategory ? categories.filter(([key]) => !childKeys.includes(key)) : categories.filter(([key]) => childKeys.includes(key));
+        setCategoriesToShow(categoriesToDisplay)
+    }, [categories])
+
+    /*
+     * Gets display name of selected category.
+     */
+    useEffect(() => {
```

Comment:
This could be moved into the other useEffect with just `categories` as a dependency. No need to have two.